### ISSUE TITLE
error on event.instance_detections2D

### ISSUE THREAD

**Issue by wcarvalho** [allenai/ai2thor#102]
When getting the 2D bounding boxes for cabinets, it tracks the door and not the cabinet itself. Here is an example where the bounding box doesn't track the cabinet. The bounding boxes are visible.

https://www.dropbox.com/s/u9hzislr1emrm3m/Screenshot%202019-01-15%2017.46.11.png?dl=0

this is a minimal example that gets you to that screen
```
from matplotlib import pyplot as plt
import ai2thor.controller

controller = ai2thor.controller.Controller()
controller.start()
controller.reset('FloorPlan28')


controller.step(
  dict(action='Initialize', 
        gridSize=0.25, 
        renderClassImage=True, 
        renderObjectImage=True
      ))
controller.step(dict(action='Teleport', x=-1.25, y=1.00, z=-1.5))
controller.step(dict(action='RotateRight'))
controller.step(dict(action='MoveAhead'))

controller.step(dict(action='MoveAhead'))
controller.step(dict(action='LookDown'))

controller.step(dict(action='MoveRight'))
controller.step(dict(action='MoveRight'))

controller.step(dict(action='MoveRight'))
controller.step(dict(action='MoveRight'))

controller.step(dict(action='LookUp'))
controller.step(dict(action='LookUp'))

event = controller.step(dict(action='MoveRight'))

controller.step(dict(action='OpenObject', objectId='Cabinet|-00.33|+01.89|-02.51'))
controller.step(dict(action='OpenObject', objectId='Cabinet|-00.33|+01.89|-03.24'))
```

**Comment by roozbehm** [comment]
We are going to fix this for all cabinets in a new version that we are going to release within a month.

**Comment by wcarvalho** [comment]
That's great. Another thing is things like sinks don't have a corresponding instance_detections2D. Will this also be fixed?

**Comment by roozbehm** [comment]
Yes, we will fix those as well.

**Comment by wcarvalho** [comment]
Hey, following up on this. It seems that many objects (e.g. sinks) still don't have instance_detections?

**Comment by winthos** [comment]
These missing boxes have been added as of version 2.4

**Comment by ekolve** [comment]
Closing - issue appears to be resolved wit 2.4.0.
